### DISCUSSION TITLE
SHA256 mismatch on fxfactory Cask 8.0.13,7767

### DISCUSSION THREAD

**Discussion by srobadelic** [Homebrew/discussions#5155] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: e5fefd73cd97cd36ae3af29551f529ae59b333d6
Last commit: 5 days ago
Core tap JSON: 24 Feb 15:02 UTC
Core cask tap JSON: 24 Feb 15:11 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.fPYJGwP7pL/org.xquartz:0
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 15.0.0 build 1500
Git: 2.44.0 => /usr/local/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.3.1-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  adoptopenjdk
  android-messages
  atom
  cakebrew
  cronnix
  komodo-edit
  obs-virtualcam
  remote-desktop-manager-free
  yo

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  mitmproxy

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  geoip
  youtube-dl

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libndi.4.dylib
  /usr/local/lib/libndi.dylib
```

### Description of issue

SHA256 Checksum is not referrencing correct hash value (or source has been modified?)
Downloaded directly from https://fxfactory.s3.amazonaws.com/noiseindustries/fxfactory/FxFactory-8.0.13-7767.zip
and confirmed 0a0adcc5be419f3f9245b7bb2f1382ee82a1c577c02613ac67513c1d6a42e908 is actual.

==> Upgrading 1 outdated package:
fxfactory 8.0.13,7761 -> 8.0.13,7767
==> Upgrading fxfactory
==> Downloading https://fxfactory.s3.amazonaws.com/noiseindustries/fxfactory/FxF
######################################################################### 100.0%
==> Purging files for version 8.0.13,7767 of Cask fxfactory
Error: fxfactory: SHA256 mismatch
Expected: 0bb9657433da0867e4196ea65a29d08ee395281784a19c242a5a9d21a45bc1ad
  Actual: 0a0adcc5be419f3f9245b7bb2f1382ee82a1c577c02613ac67513c1d6a42e908
    File: /Users/steve/Library/Caches/Homebrew/downloads/a08ebb9e9dded57fa766226fd37dffc74087ef62f88e3f71b747539d93790769--FxFactory-8.0.13-7767.zip
To retry an incomplete download, remove the file above.

### Relevant casks

fxfactory

**Comment by SMillerDev** [comment]
https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match

**Comment by chenrui333** [comment, marked as answer]
I filed an PR of updating the checksum, https://github.com/Homebrew/homebrew-cask/pull/168220

**Reply by ALiersEL** [reply]
Thanks for the fix, but the SHA256 mismatch happens again on version 8.0.16,7877. 

Here is my output:
```
==> Upgrading 1 outdated package:
fxfactory 8.0.16,7859 -> 8.0.16,7877
==> Upgrading fxfactory
==> Downloading https://fxfactory.s3.amazonaws.com/noiseindustries/fxfactory/FxFactory-8.0.16-7877.zip
################################################################################################################# 100.0%
==> Purging files for version 8.0.16,7877 of Cask fxfactory
Error: fxfactory: SHA256 mismatch
Expected: 59967d2ae46570dafa7ab4bf680e068c0b26ec909e70046e2491d61b7dd263b7
  Actual: 1f7b09867d2aa3eebcf3096e57fa38fc5d4fd8fe1e736156211845ebd8af00f1
    File: /Users/joezhu/Library/Caches/Homebrew/downloads/e82f9299678482abe077e886d46a24ba71d7c1d7d54d64b4c3f8151867725104--FxFactory-8.0.16-7877.zip
To retry an incomplete download, remove the file above.
```

I have tried to remove the file and upgrade again. Also, I have reset my homebrew, but it won't help.
Could you please fix this too? Thanks a lot.

**Reply by p-linnane** [reply]
This isn't an issue with Homebrew. What's happening is that the upstream developer is releasing a version, and at some point afterward replacing the archive for some reason without changing the version number. This is a very poor practice, and should be discussed with upstream.

**Comment by srobadelic** [comment]
All,

I have reached out to FXFactory support and asked for them to provide a revision to their product as this is throwing of SHA checks for us (their customers).

Thanks,
SS

**Comment by srobadelic** [comment]
Here is the response from FXFactory Support. It does not sound as if they will be responsive to changing this immediately, so I would think this repo should be retired as it sounds that they will continue doing this for the time being.

Hi Steven,

I apologize for the inconvenience. We received a similar report a few months ago, where we explained that our software will receive numerous unscheduled updates, sometimes just to address concerns of a single customer. This is the nature of the video industry, where we know customers are on deadlines and they are our top priority. The side effect is that within the same release day we might end up with two or more versions of the same installer. We don't do this to disrupt management tools... and hopefully we can adopt some new naming conventions to avoid this type of problem on your end. 


--
Gabe
***@***.*** ***@***.***>

**Comment by srobadelic** [comment]
Could FXFactory just be removed as they will not confirm to the standard? Or, perhaps the SHA256 value could be updated since they are acknowledging their behavior.

SS
